### PR TITLE
migrate nginx to k8s.gcr.io image

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
`test/e2e/common/node/podtemplates.go` change may be not needed. (Update it by the way. It has no side effect, I think)

#### Which issue(s) this PR fixes:

part of #108562

#### Special notes for your reviewer:
Only replaced nginx images in this PR.
Some other image has no alternative image: https://github.com/kubernetes/kubernetes/issues/108562#issuecomment-1061385476

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/testing-manifests/statefulset/nginx/statefulset.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
     spec:
       containers:
       - name: nginx
-        image: nginx:1.15-alpine
```
Comment (dims) [inline test/e2e/testing-manifests/statefulset/nginx/statefulset.yaml:18]: Do we know where we load this yaml and make the modification there? using `imageutils.GetE2EImage` in the code in addition to modifying yaml here? ( the `GetE2EImage` allows folks to inject alternate repositories )

File: test/e2e/testing-manifests/pod [hunk 2]
```diff
@@ -8,6 +8,6 @@ metadata:
 spec:
   containers:
   - name: nginx
-    image: nginx
```
Comment (dims) [inline test/e2e/testing-manifests/pod:11]: same as below.

Comment (pacoxu) [reply test/e2e/testing-manifests/pod:11]: https://github.com/kubernetes/kubernetes/blob/4e30fe40df258d30dac6d2cc01bd75f175445ac8/test/e2e_node/e2e_node_suite_test.go#L99-L100
-  In `test/e2e_node/testing-manifests`, another image need to be migrated `docker.io/nfvpe/sriov-device-plugin:v3.1`.
- In `test/e2e/testing-manifests/pod` the nginx needs to updated

File: test/e2e/testing-manifests/statefulset/nginx/statefulset.yaml (hunk 1 above)
Comment (pacoxu) [reply test/e2e/testing-manifests/statefulset/nginx/statefulset.yaml:18]: In `test/e2e/framework/manifest/manifest.go`, StatefulSetFromManifest will use it. I changed that.

**Review by liangyuanpeng — APPROVED** [review]

**Pull request merged**